### DISCUSSION TITLE
is there any way to cancel dynamic add task

### DISCUSSION THREAD

**Discussion by ibmsoft** [micronaut-projects/micronaut-core#6699] [Q&A]
for example
 use following code dynamic add task,how to cancel the added task

`
@Singleton
@Slf4j
class DynamicTaskService {

    @Inject
    ConversionService conversionService

    @Inject
    DynamicTask dynamicTask

    @Inject
    @Named(TaskExecutors.SCHEDULED)
    TaskScheduler taskScheduler

    def fixedRate(String fixedRate) {
        Optional<Duration> converted = conversionService.convert(fixedRate, Duration.class)
        Duration duration = converted.orElseThrow(() -> {
            throw new Exception("this parameter:" + fixedRate + " illegal！")
        })
        taskScheduler.scheduleAtFixedRate(Duration.ofSeconds(0), duration, dynamicTask)
    }
    
}
`

**Comment by graemerocher** [comment]
call `cancel` on the returned `ScheduledFuture` from the `scheduleAtFixedRate` method
